common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -541,24 +540,32 @@ public void copy(int srcIdx, char[] dst, int dstIdx, int length) {
             throw new NullPointerException("dst");
         }
 
-        final int thisLen = value.length;
-
-        if (srcIdx < 0 || length > thisLen - srcIdx) {
+        if (srcIdx < 0 || length > length() - srcIdx) {
             throw new IndexOutOfBoundsException("expected: " + "0 <= srcIdx(" + srcIdx + ") <= srcIdx + length("
-                            + length + ") <= srcLen(" + thisLen + ')');
+                            + length + ") <= srcLen(" + length() + ')');
         }
 
         final int dstEnd = dstIdx + length;
-        for (int i = srcIdx, j = dstIdx; j < dstEnd; i++, j++) {
-            dst[j] = b2c(value[i]);
+        for (int i = dstIdx, j = srcIdx + arrayOffset(); i < dstEnd; i++, j++) {
+            dst[i] = b2c(value[j]);
         }
     }
 
     @Override
+    public AsciiString subSequence(int start) {
```

Comment:
Maybe I'm missing something but the super.subSequence below will create a new ByteString which will cause then a ClassCastException if I don't have a brain fart:

https://github.com/netty/netty/blob/master/common/src/main/java/io/netty/util/ByteString.java#L420